python/pyspark/pandas/tests/computation/test_compute.py

```diff
@@ -101,16 +101,10 @@ def test_mode(self):
         with self.assertRaises(ValueError):
             psdf.mode(axis=2)
 
-        def f(index, iterator):
-            return ["3", "3", "3", "3", "4"] if index == 3 else ["0", "1", "2", "3", "4"]
-
-        rdd = self.spark.sparkContext.parallelize(
-            [
-                1,
-            ],
-            4,
-        ).mapPartitionsWithIndex(f)
-        df = self.spark.createDataFrame(rdd, schema="string")
+        data = [(i,) for i in ["3", "3", "3", "3", "4"]] + [
+            (i,) for i in ["0", "1", "2", "3", "4"]
+        ] * 3
+        df = self.spark.createDataFrame(data)
```

Comment:
cc @zhengruifeng FYI

Reply:
I don't think the new test can generate the expected partitioning

Reply:
I see. Then just let me revert this changes.